main.tf

```diff
@@ -14,6 +14,11 @@ provider "template" {
 
 # VPC CONFIGURATION
 
+locals {
+  vpc_public_subnets  = "${split(",", length(var.vpc_public_subnets) > 0 ? join(",", var.vpc_public_subnets) : join(",", list(cidrsubnet(var.vpc_cidr, 8, 1), cidrsubnet(var.vpc_cidr, 8, 2), cidrsubnet(var.vpc_cidr, 8, 3))))}"
```

Comment:
![dafuq](https://user-images.githubusercontent.com/3058150/56340101-23d86280-61b1-11e9-8465-79a96db8611b.jpg)

😄

Reply:
Hahaha 😅 Ugly AF but necessary -> https://github.com/hashicorp/terraform/issues/18259#issuecomment-438407005

Reply:
HCL could really benefit from being able to split functions into multiple lines... This would have been so much more easy to read:

```hcl
locals {
  vpc_public_subnets  = "${split(",",
    length(var.vpc_public_subnets) > 0
    ? join(",", var.vpc_public_subnets)
    : join(",", list(
        cidrsubnet(var.vpc_cidr, 8, 1),
        cidrsubnet(var.vpc_cidr, 8, 2),
        cidrsubnet(var.vpc_cidr, 8, 3)
      ))
  )}"
}
```